Services/communication_service.c

```diff
@@ -35,31 +53,753 @@
+          set_packet_length(packet, sizeof(int8_t) + 1);
+          if (queue_response(packet) != SATR_OK) {
+            return SATR_ERROR;
+          }
+          break;
+
+      case S_SET_CONTROL:
+          S_config.PA.status = (uint8_t)packet->data[IN_DATA_BYTE];
+          S_config.PA.mode = (uint8_t)packet->data[IN_DATA_BYTE + 1];
+          S_config.PA.status = csp_ntoh32((uint32_t)S_config.PA.status);
+          S_config.PA.mode = csp_ntoh32((uint32_t)S_config.PA.mode);
+          status = HAL_S_setControl(S_config.PA);
+          memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+          set_packet_length(packet, sizeof(int8_t) +  1);
+          if (queue_response(packet) != SATR_OK) {
+            return SATR_ERROR;
+          }
+          break;
+
+      case S_SET_ENCODER:
+          S_config.enc.scrambler = (uint8_t)packet->data[IN_DATA_BYTE];
+          S_config.enc.filter = (uint8_t)packet->data[IN_DATA_BYTE + 1];
+          S_config.enc.modulation = (uint8_t)packet->data[IN_DATA_BYTE + 2];
+          S_config.enc.rate = (uint8_t)packet->data[IN_DATA_BYTE + 3];
+          S_config.enc.scrambler = csp_ntoh32((uint32_t)S_config.enc.scrambler);
+          S_config.enc.filter = csp_ntoh32((uint32_t)S_config.enc.filter);
+          S_config.enc.modulation = csp_ntoh32((uint32_t)S_config.enc.modulation);
+          S_config.enc.rate = csp_ntoh32((uint32_t)S_config.enc.rate);
+          status = HAL_S_setEncoder(S_config.enc);
+          memcpy(&packet->data[STATUS_BYTE], &status, sizeof(int8_t));
+          set_packet_length(packet, sizeof(int8_t) +  1);
+          if (queue_response(packet) != SATR_OK) {
+            return SATR_ERROR;
+          }
+          break;
+
+      case S_SET_CONFIG:
+          cnv8_F(&packet->data[IN_DATA_BYTE], &S_config.freq);
+          S_config.freq = csp_ntohflt(S_config.freq);
+          S_config.PA_Power = (uint8_t)packet->data[IN_DATA_BYTE];//plus 4 because float takes 4B
+          S_config.PA_Power = csp_ntoh32((uint32_t)S_config.PA_Power);
+          S_config.PA.status = (uint8_t)packet->data[IN_DATA_BYTE + 1];
+          S_config.PA.mode = (uint8_t)packet->data[IN_DATA_BYTE + 2];
+          S_config.PA.status = csp_ntoh32((uint32_t)S_config.PA.status);
+          S_config.PA.mode = csp_ntoh32((uint32_t)S_config.PA.mode);
+          S_config.enc.scrambler = (uint8_t)packet->data[IN_DATA_BYTE + 3];
+          S_config.enc.filter = (uint8_t)packet->data[IN_DATA_BYTE + 4];
+          S_config.enc.modulation = (uint8_t)packet->data[IN_DATA_BYTE + 5];
+          S_config.enc.rate = (uint8_t)packet->data[IN_DATA_BYTE + 6];
+          S_config.enc.scrambler = csp_ntoh32((uint32_t)S_config.enc.scrambler);
+          S_config.enc.filter = csp_ntoh32((uint32_t)S_config.enc.filter);
+          S_config.enc.modulation = csp_ntoh32((uint32_t)S_config.enc.modulation);
+          S_config.enc.rate = csp_ntoh32((uint32_t)S_config.enc.rate);
+          status = HAL_S_setFreq (S_config.freq) +
```

Comment:
Do these return the sum of the return codes?

Reply:
Yes, it was the easiest way of knowing if there was an error (non-zero status) or not. But we could definitely improve that. If knowing what status each HAL returns is important, we can come up with an algorithm to specifies that.

Reply:
I don't mind this approach, we should document this somewhere or at least make it clear in the header comment

Reply:
Noted.